### PR TITLE
Feat: django 4.2 support

### PR DESCRIPTION
**Pull request by mogoh**
## Description

I just wanted to create a pull request for discussing progress.
Even though I am at an OK level of using git, I am not that familiar with github specific pull requests.
So, if I understood that correctly, I can still rebase the branch before merging.

This pull request is for issue #122 .

## Related resources

* #122

## Checklist

* [x] I have opened this pull request against ``master``
* [x] I have added or modified the tests when changing logic
* [x] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on 
[Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

**Comment by mogoh** [comment]
BTW: If it is considered bad, to open a pull request for this instead of integrating the changes into the existing pull request #120 , we can gladly close this one. But I don't know how to append commits to a foreign pull request.

**Comment by marksweb** [comment]
@mogoh while you've got an open PR it just shows what's happening on your branch here. You can rebase your branch and keep making changes. They'll all just show up here.

If the branch isn't ready yet, you can also have draft PRs. This gives people visibility of what you're doing but let's them know it's not ready to merge yet.

**Comment by mogoh** [comment]
@marksweb I did not know about draft pull requests, and now I changed this one into a draft. Thanks.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    autofix_prs: false
+    autoupdate_commit_msg: 'ci: pre-commit autoupdate'
+    autoupdate_schedule: monthly
+
+repos:
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v3.7.0
+    hooks:
+      - id: pyupgrade
+        args: ["--py37-plus"]
```
Comment (marksweb) [inline .pre-commit-config.yaml:15]:
```suggestion
        args: ["--py38-plus"]
```

**Review by marksweb — CHANGES_REQUESTED** [review]
This looks excellent @mogoh, thank you.

Just 1 minor thing on pyupgrade I've suggested.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (mogoh) [reply .pre-commit-config.yaml:15]: Ah, yes, sure!

**Comment by mogoh** [comment]
I guess I missed something.
Where should I increase the version number?
I have changed the `CHANGELOG.rst` but haven't changed `djangocms_picture/__init__.py`.
Should I just increase the version number to 4.1.0 and be good?

**Comment by marksweb** [comment]
@mogoh Don't worry about changing version. We have dedicated PRs to handle the version change, and the release process for this needs updating so I can do that as it needs changes on pypi as well if I've not set it up already.

**Review by marksweb — APPROVED** [review]

**Comment by marksweb** [comment]
@mogoh I should add to that, we usually reserve a header in changelogs for "unreleased" that you can then add the PR changes to, and once there are changes ready for a release, all unreleased changes get moved to a heading and date for the new version that's getting released.

Some projects use a tool called [`bump2version`](https://github.com/c4urself/bump2version) which largely automates the version change process. I might integrate this here as well.

**Pull request merged**